amd_openvx_extensions/amd_rpp/source/Spectrogram.cpp

```diff
@@ -0,0 +1,379 @@
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &data->src_desc_ptr->numDims, sizeof(data->src_desc_ptr->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, &data->in_tensor_dims, sizeof(vx_size) * data->src_desc_ptr->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DATA_TYPE, &data->in_tensor_type, sizeof(data->in_tensor_type)));
+    if (data->in_tensor_type == vx_type_e::VX_TYPE_FLOAT32)
+        data->src_desc_ptr->dataType = RpptDataType::F32;
+    data->src_desc_ptr->offsetInBytes = 0;
+
+    // Querying for output tensor
+    data->dst_desc_ptr = &data->dstDesc;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_NUMBER_OF_DIMS, &data->dst_desc_ptr->numDims, sizeof(data->dst_desc_ptr->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_DIMS, &data->out_tensor_dims, sizeof(vx_size) * data->dst_desc_ptr->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1],VX_TENSOR_DATA_TYPE, &data->out_tensor_type, sizeof(data->out_tensor_type)));
+    if (data->out_tensor_type == vx_type_e::VX_TYPE_FLOAT32)
+        data->dst_desc_ptr->dataType = RpptDataType::F32;
+    data->dst_desc_ptr->offsetInBytes = 0;
+
+    // source_description_ptr
+    data->src_desc_ptr->n = data->in_tensor_dims[0];
+    data->src_desc_ptr->h = data->in_tensor_dims[2];
+    data->src_desc_ptr->w = data->in_tensor_dims[1];
+    data->src_desc_ptr->c = 1;
+    data->src_desc_ptr->strides.nStride = data->src_desc_ptr->c * data->src_desc_ptr->w * data->src_desc_ptr->h;
+    data->src_desc_ptr->strides.hStride = data->src_desc_ptr->c * data->src_desc_ptr->w;
+    data->src_desc_ptr->strides.wStride = data->src_desc_ptr->c;
+    data->src_desc_ptr->strides.cStride = 1;
+    data->src_desc_ptr->numDims = 4;
+
+    // source_description_ptr
+    data->dst_desc_ptr->n = data->out_tensor_dims[0];
+    data->dst_desc_ptr->w = data->out_tensor_dims[1];
+    data->dst_desc_ptr->h = data->out_tensor_dims[2];
+    data->dst_desc_ptr->c = 1;
+    data->dst_desc_ptr->strides.nStride = data->dst_desc_ptr->c * data->dst_desc_ptr->w * data->dst_desc_ptr->h;
+    data->dst_desc_ptr->strides.hStride = data->dst_desc_ptr->c * data->dst_desc_ptr->w;
+    data->dst_desc_ptr->strides.wStride = data->dst_desc_ptr->c;
+    data->dst_desc_ptr->strides.cStride = 1;
+    data->dst_desc_ptr->numDims = 4;
+
+    data->sampleLength = (unsigned int *)calloc(data->src_desc_ptr->n, sizeof(unsigned int));
+    data->windowFn = (float *)calloc(data->windowLength, sizeof(float));
+
+
+// #if ENABLE_HIP
+//     if (data->deviceType == AGO_TARGET_AFFINITY_GPU)
+//         hipMalloc(&data->hip_roi_tensor_Ptr, data->src_desc_ptr->n * sizeof(RpptROI));
+// #endif
+//     data->roi_tensor_Ptr = (RpptROI *)calloc(data->src_desc_ptr->n, sizeof(RpptROI));
+    refreshSpectrogram(node, parameters, num, data);
```

Comment:
These should be replaced with createRPPHandle calls.
We need to check with the latest changes